services/spar/test-integration/Test/Spar/DataSpec.hs

```diff
@@ -274,12 +275,24 @@ testDeleteTeam = it "cleans up all the right tables after deletion" $ do
     liftIO $ tokens `shouldBe` []
   -- The users from 'user':
   do
-    let Right uref1 = fromUserSSOId ssoid1
-    mbUser1 <- runSparCass $ Data.getSAMLUser uref1
+    mbUser1 <- case veidFromUserSSOId ssoid1 of
+      Right veid ->
+        runSparCass $
+          runValidExternalId
+            Data.getSAMLUser
+            undefined -- could be @Data.lookupScimExternalId@, but we don't hit that path.
+            veid
+      Left _email -> undefined -- runSparCass . Data.lookupScimExternalId . fromEmail $ _email
```

Comment:
Why is this case undefined?

Reply:
yes, it never happens.  it's perfectly reasonable to allow this case, and the comment two lines above says how it could be implemented, but i felt that it's better to not write dead code, but rather make it clear where we don't walk.

feel free to turn it into a `fromRight` if you think it's an improvement.